spring-shell-core/src/main/java/org/springframework/shell/DefaultApplicationRunner.java

```diff
@@ -23,17 +23,24 @@
 import org.slf4j.LoggerFactory;
 
 import org.springframework.boot.ApplicationArguments;
-import org.springframework.boot.ApplicationRunner;
 import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.core.annotation.Order;
 
 /**
- * Default {@link ApplicationRunner} which dispatches to first ordered
- * {@link ShellRunner} able to handle shell.
+ * Default {@link ShellApplicationRunner} which dispatches to the first ordered {@link ShellRunner} able to handle
+ * the shell.
  *
  * @author Janne Valkealahti
+ * @author Chris Bono
  */
+@Order(DefaultApplicationRunner.PRECEDENCE)
 public class DefaultApplicationRunner implements ShellApplicationRunner {
```

Comment:
**[UNRELATED]** I would like to rename this to `DefaultShellApplicationRunner` for specificity.

Reply:
Yeah true, we could do it with this pr.